src/Shared/Recruit.Vacancies.Client/Infrastructure/QueryStore/MongoQueryStore.cs

```diff
@@ -29,7 +29,9 @@ Task IQueryStore.DeleteAsync<T>(string typeName, string key)
             var filter = filterBuilder.Eq(d => d.ViewType, typeName)
                         & filterBuilder.Eq(d => d.Id, key);
 
-            return RetryPolicy.ExecuteAsync(context => collection.DeleteOneAsync(filter), new Context(nameof(IQueryStore.DeleteAsync)));
+            return RetryPolicy.ExecuteAsync(_ => 
+                collection.DeleteOneAsync(filter),
+                new Context(nameof(IQueryStore.DeleteAsync)));
```

Comment:
async inside the lambda

Reply:
Hmmm. This is OK isn't it? `collection.DeleteOneAsync` returns a Task which satisfies `RetryPolicy.ExecuteAsync`. I don't need to do anything with the `DeleteOneAsync` result inside the lambda so I don't need to await it and I can just pass the Task back to the parent function.

Reply:
🤦‍♂️ Yes, by mistake I saw it somewhere put it on somewhere where I am awaiting two tasks and completely forgot and started putting them in single call lambdas as well. I will *remember* and change these back where they are not needed.

Reply:
Cool. This query triggered my impostor syndrome and I was about to give it all up and become a butcher.

Reply:
It craps my brains out.

Reply:
😱